teapots/common/ndk_helper/CMakeLists.txt

```diff
@@ -1,16 +1,31 @@
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wall -fno-exceptions -fno-rtti")
 
-add_library(ndk-helper STATIC
-            gestureDetector.cpp
-            gl3stub.cpp
-            GLContext.cpp
-            interpolator.cpp
-            JNIHelper.cpp
-            perfMonitor.cpp
-            sensorManager.cpp
-            shader.cpp
-            tapCamera.cpp
-            vecmath.cpp)
+add_library(NdkHelper
+  STATIC
+    gestureDetector.cpp
+    gl3stub.cpp
+    GLContext.cpp
+    interpolator.cpp
+    JNIHelper.cpp
+    perfMonitor.cpp
+    sensorManager.cpp
+    shader.cpp
+    tapCamera.cpp
+    vecmath.cpp
+)
 
-target_include_directories(ndk-helper PRIVATE
-                           ${ANDROID_NDK}/sources/android/native_app_glue)
+set_target_properties(NdkHelper
+  PROPERTIES
+    CXX_STANDARD 11
+    CXX_STANDARD_REQUIRED YES
+    CXX_EXTENSIONS NO
+)
+target_compile_options(NdkHelper
+  PUBLIC
+    -Wall -Werror -fno-exceptions -fno-rtti
```

Comment:
I still think CXX_FLAGS is the right way to do this.

Reply:
understood. Since it is norm to go target oriented, we go this "modern" way. It is like global variable vs. wrapped into object. In the end it is same or similar.

Reply:
Sorry writing this on my etcha-sketch.

Cxx_flags is a property so it can obey the target conventions.  Actually can set it in above function and lose this entirely.
A feature of using cxx_flags is it only gets added to c++files.  This is important if a different language is in a project and you want different flags per language.

Reply:
that is good to know.  I believe this project is all CPP files, so they could apply to all.
It might be good to show some variety of usage.  thx

Reply:
I agree with Dan. If your intention is for something to apply to the entire project, then it should be specified as a global, instead of relying on the implicit propagation of options through dependencies.

Reply:
@gpx1000 CXX_FLAGS is not a target_property, correct? I got trouble when trying to use it. and [this place](https://stackoverflow.com/questions/28282896/pass-cmake-cxx-flags-to-target-compile-options) seems to suggest that 2 ways to get it done: set() and target_compile_options(); set() method needs to use appending like "${CMAKE_CXX_FLAGS} then-the-new-ones". it is the old way, we should stick with new way as that link said.